hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3492,6 +3515,28 @@ func (r *HostedClusterReconciler) validateReleaseImage(ctx context.Context, hc *
 	return isValidReleaseVersion(version, currentVersion, hc.Spec.Networking.NetworkType)
 }
 
+func isProgressing(ctx context.Context, hc *hyperv1.HostedCluster) (bool, error) {
+	for _, condition := range hc.Status.Conditions {
+		switch string(condition.Type) {
+		case string(hyperv1.SupportedHostedCluster):
+		case string(hyperv1.ValidHostedClusterConfiguration):
+		case string(hyperv1.ValidReleaseImage):
+		case string(hyperv1.ReconciliationActive):
+			if condition.Status == metav1.ConditionFalse {
+				return false, fmt.Errorf("%s HostedCluster condition is false", string(condition.Type))
+			}
+		}
+	}
+
+	if hc.Status.Version == nil && hc.Spec.Release.Image != hc.Status.Version.Desired.Image {
```

Comment:
```suggestion
	if hc.Status.Version == nil || hc.Spec.Release.Image != hc.Status.Version.Desired.Image {
```

Reply:
Used the wrong syntax here :man_facepalming: